app/views/contexts/preview_context.py

```diff
@@ -0,0 +1,63 @@
+from typing import Generator, Mapping, Optional, Union
+
+from app.data_models import AnswerStore, ListStore, ProgressStore, QuestionnaireStore
+from app.data_models.metadata_proxy import MetadataProxy
+from app.questionnaire import QuestionnaireSchema
+from app.questionnaire.location import Location
+from app.views.contexts import Context
+from app.views.contexts.section_preview_context import SectionPreviewContext
+
+
+class PreviewContext(Context):
+    def __init__(
+        self,
+        language: str,
+        schema: QuestionnaireSchema,
+        answer_store: AnswerStore,
+        list_store: ListStore,
+        progress_store: ProgressStore,
+        metadata: Optional[MetadataProxy],
+        response_metadata: Mapping,
+        questionnaire_store: QuestionnaireStore,
```

Comment:
Don't pass a questionnaire store. Only pass what you need, this already passes, list, answer, progress, metadata etc already...

Reply:
Ok, passing only what I need in the classes now.